### DISCUSSION TITLE
How to install .deb?

### DISCUSSION THREAD

**Discussion by mariomadproductions** [86Box/86Box#2344] [Q&A] [answered]
If I have a piece of closed source x86 linux software that comes as a deb, what's the proper way to install/use it with 86box?

**Comment by Dizzy611** [comment, marked as answer]
First, you must create the universe.

No but seriously, this is going to be a long and involved process. 

Do you know which version of Debian your .deb file was built for?

Because you'll want to download an ISO of that version, the i386 ISO specifically.

You're going to want to use a machine in 86box that can boot from CDs (probably a Socket 7 Pentium of some sort). Configure your machine with enough RAM to run the version of Debian you're using, a CD drive, a hard disk big enough to hold the debian environment and your application, a network card (I suggest NE2000), and a video card that is supported (I suggest using VGA to start with), then load the ISO and install Debian. 

Then transfer the .deb file into the virtual machine either by downloading it over the network or by putting it in a floppy or cd image, and install the .deb from within the emulated version of Linux.

I will also say however that unless the .deb in question absolutely needs to be run in a cycle-accurate environment, virtualization (qemu, virtualbox, and the like) are probably sufficient for and more suited to the task of running old closed source linux software.

**Comment by re9177** [comment]
in other words: run in emulated machine dpkg -i deb filename

**Comment by mariomadproductions** [comment]
You know what, I think I was sleep deprived when I made this discussion. I meant to post it in the Box86 github, lol. For Box86, the answer is probably a bit less obvious due to interactions between native and emulated binaries. Thanks for the answers though, despite my obvious question. You might like to know I do use 86Box for old windows 98 software, which is a more appropriate use :p

**Reply by dgurney** [reply]
No worries, happens to the best of us!
